packages/lab/src/date-input/DateInput.tsx

```diff
@@ -5,52 +5,76 @@ import {
   FocusEvent,
   forwardRef,
   InputHTMLAttributes,
-  Ref,
+  KeyboardEvent,
+  ReactNode,
+  RefObject,
+  useEffect,
   useState,
 } from "react";
 import { useComponentCssInjection } from "@salt-ds/styles";
 import { useWindow } from "@salt-ds/window";
 
 import dateInputCss from "./DateInput.css";
-import { makePrefixer, useControlled, useFormFieldProps } from "@salt-ds/core";
-import { DateFormatter } from "@internationalized/date";
+import { makePrefixer, useFormFieldProps } from "@salt-ds/core";
+import {
+  CalendarDate,
+  DateFormatter,
+  DateValue,
+  getLocalTimeZone,
+  parseAbsoluteToLocal,
+} from "@internationalized/date";
+import { useDatePickerContext } from "../date-picker/DatePickerContext";
+import { getCurrentLocale } from "../calendar/internal/utils";
 
 const withBaseName = makePrefixer("saltDateInput");
-
 const isInvalidDate = (value: string) =>
   // @ts-ignore evaluating validity of date
   value && isNaN(new Date(value));
+const createDate = (date: string): Date | null => {
+  if (!date) return null;
+  try {
+    return isInvalidDate(date) ? null : new Date(date);
+  } catch (err) {
+    return null;
+  }
+};
 
-const defaultDateFormatter = (input: string): string => {
-  const date = new Date(input);
+function getCalendarDate(inputDate: string) {
+  const date = createDate(inputDate);
+  if (!date) return undefined;
+  const isoDate = parseAbsoluteToLocal(date?.toISOString());
+  return new CalendarDate(isoDate.year, isoDate.month, isoDate.day);
+}
+
+const getDateValidationStatus = (value: string | undefined) =>
+  value && isInvalidDate(value) ? "error" : undefined;
```

Comment:
why do we set to 'undefined' here if date is not valid, in the try / catch example we set to 'null' - there a reason for the difference ?

Reply:
just the intention of each value, 
the null is set on the date where we want to indicate explicitly the date value is empty
this is the validationStatus that indicates if we need to add a error state to the input, if there is no "error" state, i treated it as "an error state hasn't been set" 
not sure if the explanation makes sense XD